Makefile

```diff
@@ -181,7 +182,7 @@ build-push-e2e-catalog: ## Build the testdata catalog used for e2e tests and pus
 test-e2e: KIND_CLUSTER_NAME := operator-controller-e2e
 test-e2e: KUSTOMIZE_BUILD_DIR := config/overlays/e2e
 test-e2e: GO_BUILD_FLAGS := -cover
```

Comment:
```suggestion
test-e2e: GOOS=linux
test-e2e: GOARCH=amd64
test-e2e: GO_BUILD_FLAGS := -cover
```

Reply:
all the GOOS, GOARCH suggestions are needed in the case where the local platform is not Linux/AMD64 then the tests and image reg binaries will still work inside Kind cluster (which needs Linux/AMD64)

Reply:
We should be careful here; @joelanford had some changes in the Makefile to do the proper thing, we should emulate that, where possible.

Reply:
This _should_ be the right place to set these variables; _not_ in the `image-registry` recipe.

Reply:
Updated